### PR TITLE
Document causes of linking errors and how to fix them.

### PR DESCRIPTION
**Pull request by sjrd**
Related to https://github.com/scala-js/scala-js/pull/3751

### CODE REVIEW

File: doc/project/linking-errors.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+```
+[info] Fast optimizing .../helloworld/target/scala-2.12/helloworld-fastopt.js
+[success] Total time: 3 s, completed Sep 13, 2019 1:44:25 PM
+```
+
+
+## Using blocking APIs (e.g., `Await.result`)
+
+The JavaScript platform does not support blocking at all.
+Consequently, trying to use block APIs such `Await.result` is not supported, and will result in linking error.
```
Comment (gzm0) [inline doc/project/linking-errors.md]: blocking?

Comment (gzm0) [reply doc/project/linking-errors.md]: result in *a* linking error?

File: doc/project/linking-errors.md [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+If the `Referring to non-existent` line mentions `CompletionLatch`, it is probably related to trying to block on a `Future`.
+
+### Solution
+
+Always use asynchronous combinators, such as `map`, `filter`, `flatMap`, `foreach`, and/or for comprehensions, when using `Future`.
+Do not use `scala.concurrent.Await`.
+
+
+## Using unsupported JDK libraries (possibly transitively)
```
Comment (gzm0) [inline doc/project/linking-errors.md]: It might be helpful to make an example of a transitive problem (e.g. trying to use `scala.io`)

**Comment by sjrd** [comment]
Updated.

**Pull request merged**